File: plugins/2InjectRequest.rb
def proxenet_request_hook(request_id, request)
  @CRLF = "\r\n"
  request.sub(@CRLF*2, @CRLF+"X-Injected-By: ruby/proxenet"+@CRLF*2)
end

def proxenet_response_hook(response_id, response)
  puts "From Ruby, in response #{response_id}"
  response 
end


if __FILE__ == $0
  puts "In main"
  @rid = 11
  @req  = "GET / HTTP/1.1\r\nHost: foo\r\n\r\n"
  puts proxenet_request_hook(@rid, @req)
end

File: plugins/2InjectRequest.pl
sub proxenet_request_hook {
    my ($rid, $req) = @_;
    $crlf = "\r\n";
    $end = $crlf . $crlf;
    
    $req =~ s/$end/${crlf}X-Injected-By: perl\/proxenet${end}/;
    return $req
}

sub proxenet_response_hook {
    my ($rid, $res) = @_;
    return $res;
}


$rid = 14;
$req = "GET / HTTP/1.1\r\nHost: foo\r\n\r\n";

print proxenet_request_hook($rid, $req);
